mailet/api/src/test/java/org/apache/mailet/AttributeValueTest.java

```diff
@@ -0,0 +1,278 @@
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnEmptyListAttributeValueWhenEmptyArray() throws Exception {
+        AttributeValue<?> expected = AttributeValue.of(ImmutableList.of());
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"CollectionSerializer\",\"value\": []}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnListAttributeValueWhenArray() throws Exception {
+        AttributeValue<?> expected = AttributeValue.of(ImmutableList.of(AttributeValue.of("first"), AttributeValue.of("second")));
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"CollectionSerializer\",\"value\":[{\"serializer\":\"StringSerializer\",\"value\":\"first\"},{\"serializer\":\"StringSerializer\",\"value\":\"second\"}]}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnLongAttributeValueWhenLong() throws Exception {
+        AttributeValue<Long> expected = AttributeValue.of(42L);
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"LongSerializer\",\"value\":42}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnFloatAttributeValueWhenFloat() throws Exception {
+        AttributeValue<Float> expected = AttributeValue.of(1.0f);
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"FloatSerializer\",\"value\":1.0}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnDoubleAttributeValueWhenDouble() throws Exception {
+        AttributeValue<Double> expected = AttributeValue.of(1.0);
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"DoubleSerializer\",\"value\":1.0}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnQueueSerializableAttributeValueWhenQueueSerializable() throws Exception {
+        AttributeValue<QueueSerializable> expected = AttributeValue.of(new TestQueueSerializable(42));
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"QueueSerializableSerializer\",\"value\":{\"factory\":\"org.apache.mailet.AttributeValueTest$TestQueueSerializable$Factory\",\"value\":{\"serializer\":\"IntSerializer\",\"value\":42}}}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    void fromJsonStringShouldReturnUrlAttributeValueWhenUrl() throws Exception {
+        AttributeValue<URL> expected = AttributeValue.of(new URL("https://james.apache.org/"));
+
+        AttributeValue<?> actual = AttributeValue.fromJsonString("{\"serializer\":\"UrlSerializer\",\"value\": \"https://james.apache.org/\"}");
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
```

Comment:
Can you write a test combining a broken JSON and `fromJson` ?

Can you write a from JSON test with an unknown serializer field?

Can you write a fromJson test combining a Text serializer field and an Integer value? (idem for all serializers)